msalautomationapp/build.gradle

```diff
@@ -50,26 +50,60 @@ android {
         }
     }
 
-    testBuildType = "release"
+    testBuildType = "debug"
 
     lintOptions {
         abortOnError false
     }
 
-    flavorDimensions "main"
+    flavorDimensions "main", "broker"
 
     productFlavors {
         local {
+            dimension "main"
             applicationIdSuffix ".local"
             versionNameSuffix "-local"
             resValue("string", "application_name", "msal-local")
         }
         dist {
+            dimension "main"
             // Keep .local because the redirect url we registered on the portal contains .local, not .dist
             applicationIdSuffix ".local"
             versionNameSuffix "-dist"
             resValue("string", "application_name", "msal-dist")
         }
+
+        BrokerMicrosoftAuthenticator {
+            dimension "broker"
+        }
+
+        BrokerCompanyPortal {
+            dimension "broker"
+        }
+
+        BrokerHost {
+            dimension "broker"
+        }
+    }
+
+    variantFilter { variant ->
+        def flavorName = variant.flavors*.name
```

Comment:
How does this work? Does it matter what flavor/build the automation test app is built with?

Should we use the signature hash of the APKs instead to see if the combination is valid?

Reply:
All we are doing here is excluding specific build variants from gradle because such variants are just not supported under any circumstance. For instance, we don't support using Company Portal with a `localDebug` build of MSAL because MSAL will just not talk to it.

Reply:
So we aren't checking here if a specific flavor of MSAL can work with a specific flavor of broker. All we are doing here is just excluding some variants that we don't support in the MSAL Automation App (at least as of today).

Reply:
what I meant is that I can technically upload MSAL/Broker with any build/flavor and pick the 'right' build/flavor for the msalautomationapp, right?

Reply:
Broker yes.....MSAL is selected by the build variant (local vs dist).

So yes technically you can select Broker flavor as BrokerHost and instead supply Authenticator app when running the automation. In that case the tests will fail because the tests will be looking for a BrokerHost apk and if they don't find it then they will fail.